src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -46,7 +50,15 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event)
             $subject = null;
 
             if ($subjectRef) {
-                if (\is_array($subjectRef)) {
+                if ($subjectRef instanceof Expression) {
+                    $this->expressionLanguage ??= class_exists(Expression::class)
+                        ? new ExpressionLanguage()
+                        : throw new LogicException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed. Try running "composer require symfony/expression-language".')
+                    ;
+                    $subject = $this->expressionLanguage->evaluate($subjectRef, [
+                        'args' => $arguments,
+                    ]);
+                } elseif (\is_array($subjectRef)) {
```

Comment:
do we want to accept `new Expression` nested into `$subjectRef`? (one level only)

Reply:
Was fairly simple to implement, now expressions can be used in arrays, eg:

```php
#[IsGranted(
    attributes: new Expression('user === subject["author"]'),
    subject: [
        'author' => new Expression('args["post"].getAuthor()'),
        'post' => 'post',
    ],
)]
public function index(Post $post)
{
}
```